tensorflow/python/framework/test_util.py

```diff
@@ -266,6 +266,13 @@ def setUp(self):
     self._ClearCachedSession()
     random.seed(random_seed.DEFAULT_GRAPH_SEED)
     np.random.seed(random_seed.DEFAULT_GRAPH_SEED)
+    # Note: The following line is necessary because some test methods may error
+    # out from within nested graph contexts (e.g., via assertRaises and
+    # assertRaisesRegexp), which may leave ops._default_graph_stack non-empty
+    # under certain versions of Python. That would cause
+    # ops.reset_default_graph() to throw an exception if the stack were not
+    # cleared first.
```

Comment:
I'm still confused on why `ops._default_graph_stack` can be non-empty after an exception, since the `finally:` block in the default graph context manager should still execute, right?

If not, is it possible we have a bug in that context manager?